ipaserver/plugins/cert.py

```diff
@@ -1046,6 +1054,84 @@ def _principal_name_matches_principal(name, principal_obj):
     return principal in principal_obj.get('krbprincipalname', [])
 
 
+def _validate_san_ips(generalnames):
+    """
+    Raise a ValidationError if the subjectAltName in a CSR includes
+    any IP address(es) that do not match a DNS name in the SAN.
+
+    """
+    san_dns_ips = set()
+    for name in generalnames:
+        if isinstance(name, cryptography.x509.general_name.DNSName):
+            san_dns_ips.update(_san_dnsname_ips(name.value))
+    for name in generalnames:
+        if isinstance(name, cryptography.x509.general_name.IPAddress):
+            if unicode(name.value) not in san_dns_ips:
+                raise errors.ValidationError(
+                    name='csr',
+                    error=_(
+                        "IP address in subjectAltName (%s) does not "
+                        "match any DNS name"
+                    ) % name.value
+                )
+
+
+def _san_dnsname_ips(dnsname, dnsname_is_cname=False):
```

Comment:
The check takes at least two additional API calls for each IP addresses, which is rather expensive.

Reply:
It's such a niche case, I wouldn't really worry about it.  If there's an obvious way to cut down the number of requests, for sure let's do it, but IMO this is not a blocker.